hrl/wrappers/antmaze_wrapper.py

```diff
@@ -0,0 +1,145 @@
+from copy import deepcopy
+
+import numpy as np
+import torch
+
+from hrl.wrappers.gc_mdp_wrapper import GoalConditionedMDPWrapper
+
+
+class D4RLAntMazeWrapper(GoalConditionedMDPWrapper):
+	def __init__(self, env, start_state, goal_state, use_dense_reward=False):
+		self.env = env
+		self.norm_func = lambda x: np.linalg.norm(x, axis=-1) if isinstance(x, np.ndarray) else torch.norm(x, dim=-1)
+		self.reward_func = self.dense_gc_reward_func if use_dense_reward else self.sparse_gc_reward_func
+		self._determine_x_y_lims()
+		super().__init__(env, start_state, goal_state)
+
+	def state_space_size(self):
+		return self.env.observation_space.shape[0]
+	
+	def action_space_size(self):
+		return self.env.action_space.shape[0]
+	
+	def sparse_gc_reward_func(self, states, goals, batched=False):
+		"""
+		overwritting sparse gc reward function for antmaze
+		"""
+		# assert input is np array or torch tensor
+		assert isinstance(states, (np.ndarray, torch.Tensor))
+		assert isinstance(goals, (np.ndarray, torch.Tensor))
+
+		# TODO: change this to batched operation
```

Comment:
Delete TODO, I think it has been done :)

Reply:
Ah, I forgot about this one. 

I was actually hoping (at some point) we could remove the `batched` argument in this function, and make sure that all input `states` and `goals` are batched.

Reply:
Agreed! Everything should always be batched and a single state should have dim (1, D) rather than (D,)